backend/internal/database/book.go

```diff
@@ -357,3 +357,123 @@ func (db *BookDatabase) FetchBookAnalytics(ctx context.Context, id string, dRang
 	return analytics, err
 
 }
+
+/*
+ * Returns an array containing the favorite books for the given userID
+ */
+func (db *BookDatabase) FetchFavorites(ctx context.Context, userID string) ([]models.Book, error) {
+
+	var favoriteBooks []models.Book
+	favoriteBooks = make([]models.Book, 0)
```

Comment:
you can combine these into `favoriteBooks := make([]models.Book, 0)`

Actually, you shouldn't need to call make here at all, append will make the array if it's not constructed yet.